bakerytest/bakerytest.go

```diff
@@ -286,8 +286,10 @@ func (d *InteractiveDischarger) FinishInteraction(w http.ResponseWriter, r *http
 		httprequest.WriteJSON(w, code, body)
 		return
 	}
+	discharge.err = err
+	discharge.cavs = cavs
 	select {
-	case discharge.c <- err:
+	case discharge.c <- struct{}{}:
```

Comment:
I think I'd prefer for the error and caveats to be sent on the channel rather than set in the discharge object because there's no necessary reason that FinishInteraction might not be called twice concurrently, which would cause a race condition here.